### DISCUSSION TITLE
The display order (?) of slider anchors, editor ui and selection box controls

### DISCUSSION THREAD

**Discussion by cihe13375** [ppy/osu#30008] [Ideas]
Think about the following three scenarios (every of them happens to me at least once):
1) slider anchors are covered by the timeline and are unable to reach (note that the slider in question is not offscreen):
![1](https://github.com/user-attachments/assets/d029f4f2-262e-49b3-9d34-1fc57088255a)
2) slider anchors are covered by the control buttons of the selection box and are unable to see (surprisingly they still respond to dragging but does not respond to right click):
![2](https://github.com/user-attachments/assets/abe5fbe8-b624-47e7-aac5-a3ca8b1aeb66)
3) the selection box anchors are covered by the timeline or the progress bar, and are unable to reach:
![3](https://github.com/user-attachments/assets/3bf5058f-95e9-4273-8ecf-1f5334922dd3)

I consider 1) as a deficiency compared to stable, because stable display slider anchors above the timeline and the progress bar, so they still can be operated in this scenario. However, idk if it's good to replicate this behavior in lazer, because in lazer the slider anchor is larger while the objects in the timeline are smaller, so there is a higher change of them interfering with objects in the timeline.

point 3) is similar to 1) though it's lazer only.

I'm not sure about 2). Theoretically possible solutions include:
1) display slider anchors above those buttons. This should be an improvement because slider anchors are smaller and cannot be selected/operated by the keyboard (in construct to the transform buttons);
2) move buttons to elsewhere in such scenarios, just like how they are already avoiding the timeline / progress bar;
3) consider it as a design and do not change. This is also kinda usable because they still respond to drag (just need some luck to "guess" their position underneath the button).

**Comment by peppy** [comment]
I see this being solved in all instances by adding playfield zoom.
